pixinsight: init at 1.8.8-12

sheepforce opened the pull request:
###### Motivation for this change
Addition of the PixInsight image processing software for astrophotography. See also https://discourse.nixos.org/t/faking-root-in-build-sandbox/17089/10?u=sheepforce

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

IvarWithoutBones commented on pkgs/applications/graphics/pixinsight/default.nix:
```diff
@@ -0,0 +1,127 @@
+, mime-types, libGL, libpulseaudio, alsa-lib, nss, gd, gst_all_1, nspr, expat, fontconfig
+, dbus, glib, zlib, openssl, libdrm, cups, avahi-compat, xorg, wayland, libudev0-shim
+# Qt 5 subpackages
+, qt5
```
Since you used `libsForQt5.callPackage`, the `qt5` subpackages should already be in scope:
```suggestion
, qtbase, qtgamepad, qtserialport, qtserialbus, qtvirtualkeyboard, qtmultimedia, qtwebkit
```

IvarWithoutBones commented on pkgs/applications/graphics/pixinsight/default.nix:
```diff
@@ -0,0 +1,127 @@
+  pname = "pixinsight";
+  version = "1.8.8-12";
+
+  src = requireFile rec {
```
Could you add the `message` attribute, explaining where to get this file? For example, see https://github.com/NixOS/nixpkgs/blob/1b1fe46bb1f2d86cd47353c8d921f6c09677b294/pkgs/games/sm64ex/default.nix#L13

Might be nice to explain how to add it to the store as well.

IvarWithoutBones commented on pkgs/applications/graphics/pixinsight/default.nix:
```diff
@@ -0,0 +1,127 @@
+  ];
+  dontWrapQtApps = true;
+  postFixup = ''
+    wrapProgram $out/opt/PixInsight/bin/PixInsight ${builtins.toString qtWrapperArgs}
```
```suggestion
    wrapProgram $out/opt/PixInsight/bin/PixInsight "''${qtWrapperArgs[@]}"
```

sheepforce replied:
This leads to an error unfortunately. `makeWrapper doesn't understand the arg "QTWEBENGINEPROCESS_PATH"`. Would it be okay to leave it as is? I also find the nix string interpolation more readable.

IvarWithoutBones replied:
Sounds good :+1:

IvarWithoutBones commented on pkgs/applications/graphics/pixinsight/default.nix:
```diff
@@ -0,0 +1,127 @@
+    wrapProgram $out/opt/PixInsight/bin/PixInsight ${builtins.toString qtWrapperArgs}
+  '';
+
+  meta = with lib; {
```
Could you add `meta.mainProgram`?

IvarWithoutBones reviewed (approved):
This LGTM :+1:

sheepforce committed:
pixinsight: init at 1.8.8-12


pixinsight: review cleanup


pixinsight: review cleanup


pixinsight: switch back to working wrapper
